src/main/helpers/RouterHelpers.ts

```diff
@@ -37,17 +37,17 @@ export const returnNextPage = (req: AppRequest, res: Response, redirectUrl: stri
 
 export const returnValidUrl = (redirectUrl: string, validUrls?: string[]): string => {
   validUrls = validUrls ?? Object.values(PageUrls);
+
+  // Parse the redirectUrl to get the pathname without query parameters
+  const parsedUrl = new URL(redirectUrl, 'http://example.com'); // Base URL is required for parsing
```

Comment:
How does it behave, if it is a secure url like https://example.com ?

Reply:
Locally it works with https urls, this may flag another fortify issue but we will have to deal with them as they arise.